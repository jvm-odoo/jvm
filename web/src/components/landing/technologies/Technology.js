import React from 'react'
import styled from 'styled-components'

const TechnologyContainer = styled.div`
    margin-top: 48px;
    margin-right: 24px;
    position: relative;

    :hover {
        div {
            visibility: visible;
            opacity: 1;
        }
    }
`
const TechnologyImage = styled.img`
    height: 70px;
    width: 70px;
`

const TechnologyTooltipContainer = styled.div`
    visibility: hidden;
    display: flex;
    height: 70px;
    width: 70px;
    justify-content: center;
    align-items: center;
    position: absolute;
    opacity: 0;
    transition: 0.2s;
`

const TechnologyTooltip = styled.span`
    cursor: default;
    padding: 5px;
    background-color: rgba(0, 0, 0, 0.6);
    color: #FFF;
    border-radius: 5px;
`

export const Technology = ({ image, name }) => (
    <TechnologyContainer>
        <TechnologyTooltipContainer>
            <TechnologyTooltip>{name}</TechnologyTooltip>
        </TechnologyTooltipContainer>
        <TechnologyImage src={image} alt={name} />
    </TechnologyContainer>
)