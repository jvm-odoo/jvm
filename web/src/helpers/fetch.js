export const fetchGet = async (url) => {
  const response = await fetch(url)
  return response.json()
}
