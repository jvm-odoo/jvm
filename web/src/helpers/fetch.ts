export const fetchGet = async (url: string) => {
    const response = await fetch(url)
    return response.json()
}

export const fetchPost = async (url: string, data: any, token?: string) => {
    const headers: Headers = new Headers({
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    })

    if (token) {
        headers.append('Authorization', `Token ${token}`)
    }

    const response = await fetch(url, {
        method: 'POST',
        headers,
        mode: 'cors',
        body: JSON.stringify(data)
    })

    return response.json()
}