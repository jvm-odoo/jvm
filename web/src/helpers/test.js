import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(Component)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        {Component}
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}
