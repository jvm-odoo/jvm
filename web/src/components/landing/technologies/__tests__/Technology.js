import React from 'react'

import { Technology } from '../Technology'
import { shallowMatchSnapshot } from '../../../../helpers/test'

it('renders correctly', () => shallowMatchSnapshot(<Technology />))
