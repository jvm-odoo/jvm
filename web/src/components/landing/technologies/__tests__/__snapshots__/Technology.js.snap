// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.span />
  </styled.div>
  <styled.img />
</styled.div>
`;
